purchase_all_shipments/models/purchase_order.py

```diff
@@ -0,0 +1,44 @@
+# Copyright 2018 Camptocamp SA
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl)
+from odoo import fields, models
+
+
+class PurchaseOrder(models.Model):
+    _inherit = "purchase.order"
+
+    all_picking_ids = fields.One2many(
+        "stock.picking", string="All Pickings", compute="_compute_all_pickings"
+    )
+    all_picking_count = fields.Integer(
+        "All Pickings Count", compute="_compute_all_picking_count"
+    )
+
+    def _compute_all_picking_count(self):
+        for rec in self:
+            rec.all_picking_count = len(rec.all_picking_ids)
+
+    def _compute_all_pickings(self):
+        for rec in self:
+            groups = rec.mapped("picking_ids.group_id")
+            all_picking_ids = self.env["stock.picking"].search(
+                [("group_id", "in", groups.ids)]
+            )
+            rec.all_picking_ids = all_picking_ids
+
+    def action_view_all_pickings(self):
+        """Similar to the view_picking method in the purchase module"""
+        action_data = self.env.ref("stock.action_picking_tree_all").read()[0]
```

Comment:
This will raise `odoo.addons.base.models.ir_model: Access Denied by ACLs for operation: read, uid: 6, model: ir.actions.act_window`:
```
 odoo.http: You are not allowed to access 'Action Window' (ir.actions.act_window) records.

This operation is allowed for the following groups:
        - Administration/Settings

Contact your administrator to request access if necessary.
```
cf. https://github.com/odoo/odoo/blob/15.0/addons/purchase_stock/models/purchase.py#L123 to update the method